KC7-Foundation/kc7_scoreboard
# Add User Profile Page with Session and Event Details

Resolves Issue #31

### Description
This pull request introduces a User Profile page that displays users' sessions and events attended, along with their corresponding scores. The profiles are now linked to the users on the leaderboard and the manager users page. Additionally, I have updated the icon for the leaderboard on the sidebar to ensure better visual consistency with the other icons.

### How to Test
1. Use production data for testing.
2. Log in using an admin account.
3. Perform the following checks:
   - Verify the presence of the "My Performance" tab on the sidebar.
   - Click on usernames displayed on the Leaderboard page and ensure they lead to the corresponding user profiles.
   - Click on usernames displayed on the Manage Users page and ensure they also lead to the corresponding user profiles.
4. To test error handling, try accessing a route with an invalid username and verify that a 404 error is returned.

### Screenshots
The following screenshots demonstrate the changes:

| Session names and scores | New 'My Performance' tab and new Leaderboard icon |
| --- | --- |
| ![Session names and scores](https://github.com/KC7-Foundation/kc7_scoreboard/assets/97637517/6bebfafa-ee53-41f3-b2ca-274a9aee9f7c) | ![New My Performance tab and new Leaderboard Icon](https://github.com/KC7-Foundation/kc7_scoreboard/assets/97637517/664dc10e-80f0-45b0-9838-888dd07907c4) |
